### PR TITLE
inserted a comment at top of Begin.f90 to verify PR

### PR DESCRIPTION
**Pull request by rniemeyer07**

### CODE REVIEW

**Comment by rniemeyer07** [comment]
This is the first stage in integrating my version of RBM with reservoirs with the current version of RBM. I mainly added variables to read in and calculate reservoir variables, and created the loop in Systmm.f90 for the code to know when do "normal" RBM river calculations or reservoir calculations.

File: src/Read_Forcing.f90 [hunk 1]
```diff
@@ -68,8 +78,9 @@ SUBROUTINE Read_Forcing
 !
   Q_trib(nr)=Q_out(no_heat)    
   nrec_heat=heat_cells*(ndays-1)+no_heat
-  read(36,'(i5,2f6.1,2f7.4,f6.3,f7.1,f5.1)' &
-         ,rec=nrec_heat) ncell &
+!  read(36,'(i5,2f6.1,2f7.4,f6.3,f7.1,f5.1)' &
+!         ,rec=nrec_heat)
```
Comment (YifanCheng) [inline src/Read_Forcing.f90]: Recommend to delete two comment lines (L39, L41, L81 and L82)

File: src/Read_Forcing.f90 [hunk 2]
```diff
@@ -19,17 +19,27 @@ SUBROUTINE Read_Forcing
...
     nrec_heat=heat_cells*(ndays-1)+no_heat
+   ! print *, 'no_heat', no_heat,'nrec_flow',nrec_flow, 'nrec_heat', nrec_heat
 !
-    read(35,'(2i5,2f10.1,2f6.1,f7.1,f6.2)' &
-           ,rec=nrec_flow) nnd,ncell &
-           ,Q_out(no_heat),Q_dmmy,Q_diff(no_heat) &  
-           ,depth(no_heat),width(no_heat),u(no_heat)
+!    read(35,'(i1,1x,i1,1x,f5.1,1x,f4.1,1x,f2.1,1x,f2.1,1x,f3.1,1x,f1.2)' &
+    read(35,*) &
+    !  , rec=nrec_flow) &
```
Comment (YifanCheng) [inline src/Read_Forcing.f90:26]: Recommend to delete the two comment line (L24 and L26)

File: src/Read_Forcing.f90 [hunk 3]
```diff
@@ -19,17 +19,27 @@ SUBROUTINE Read_Forcing
 !
     nrec_flow=flow_cells*(ndays-1)+no_flow
     nrec_heat=heat_cells*(ndays-1)+no_heat
+   ! print *, 'no_heat', no_heat,'nrec_flow',nrec_flow, 'nrec_heat', nrec_heat
 !
```
Comment (YifanCheng) [inline src/Read_Forcing.f90:23]: Delete comment line 22. This is used for testing.

File: src/Read_Forcing.f90 [hunk 4]
```diff
@@ -19,17 +19,27 @@ SUBROUTINE Read_Forcing
...
+    !  , rec=nrec_flow) & 
+           nnd,ncell &
+           ,Q_out(no_heat),Q_dmmy, Q_diff(no_heat) &
+           ,depth(no_heat) &
+           ,width(no_heat) &
+           ,u(no_heat)
 !
+
+!print *, 'nnd', nnd, 'ncell',ncell,'Q_out',Q_out(no_heat),'Q_dmy',Q_dmmy & 
+!,'Qdiff',Q_diff(no_heat),'dep',depth(no_heat),'width',width(no_heat),'u',u(no_heat)
```
Comment (YifanCheng) [inline src/Read_Forcing.f90:35]: Delete comment line 34 and 35. This is used for testing.

File: src/Block_Reservoir.f90 [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        !       reservoir to RBM variables
+        !----------------------------------------------------------------------
+
+        !------------------------read in reservoir info--------------------------
+        real, dimension (:), allocatable:: dam_lat, dam_lon, res_grid_lat, res_grid_lon
+        real, dimension (:), allocatable:: res_depth_feet, res_width_feet, res_length_feet
+        integer, dimension (:), allocatable:: dam_number, start_operating_year
+        real, dimension (:), allocatable:: res_top_vol, res_bot_vol, res_max_flow, res_min_flow
+        integer, dimension (:), allocatable:: res_start_node, res_end_node
+       !  integer, dimension (:,:), allocatable:: nodes_x   !for each reach, what are all the nodes
```
Comment (YifanCheng) [inline src/Block_Reservoir.f90]: Delete comment line 15. If node_x is not used, there is no need to keep it there. Same for line 26, 57 and 58

File: src/rbm10_VIC.f90 [hunk 6]
```diff
@@ -94,14 +99,34 @@ PROGRAM RBM10_VIC
 !
 !     Open file with hydrologic data
 !
-open(unit=35,FILE=TRIM(flow_file) ,FORM='FORMATTED',ACCESS='DIRECT' ,RECL=60,STATUS='old')
+!open(unit=35,FILE=TRIM(flow_file) ,FORM='FORMATTED',ACCESS='DIRECT' ,RECL=60,STATUS='old')
```
Comment (YifanCheng) [inline src/rbm10_VIC.f90]: Remove line 102 as well as line 110. No need to keep the old code there

File: src/Begin.f90 [hunk 7]
```diff
@@ -68,6 +70,50 @@ Subroutine BEGIN(param_file,spatial_file)
  allocate(head_cell(nreach))
  allocate(segment_cell(nreach,ns_max))
  allocate(x_dist(nreach,0:ns_max))
+! Allocate reservoir info
```
Comment (YifanCheng) [inline src/Begin.f90:73]: Format this comment. Make it similar to John's comment style. Similar comment to line 99

File: src/Begin.f90 [hunk 8]
```diff
@@ -68,6 +70,50 @@ Subroutine BEGIN(param_file,spatial_file)
...
+              , res_top_vol(nreservoir) &
+              , res_bot_vol(nreservoir),res_depth_feet(nreservoir) &
+              , res_width_feet(nreservoir), res_length_feet(nreservoir) &
+              ,res_start_node(nreservoir), res_end_node(nreservoir) &
+              , res_min_flow(nreservoir)
+
+       print *, 'nreservoir', nreservoir, 'dam_num', dam_number(nreservoir) &
+    , 'end', res_end_node(nreservoir) , 'start', res_start_node(nreservoir)
+    end do
+end if
```
Comment (YifanCheng) [inline src/Begin.f90:115]: From line 101 to 114, can you make coding style consistent throughout the entire code?
For example, in line 109, it is better to have a blank between the comma and `res_start_node`.
Another example, when one line is too long, you switch to a new line, how many space do you use? For line 105, you use 6 blanks but for line 113, you even move that line forward three blank space.

File: src/Begin.f90 [hunk 9]
```diff
@@ -137,8 +183,27 @@ Subroutine BEGIN(param_file,spatial_file)
...
+           else
+             res_presx = .false.
+           end if
+
+        !      X,Y matrix with nodes in each reach
+            nodes_x(nr,ncell) = node    ! matrix of nodes for each cell
+            res_num(nr,ncell) = node_res   ! matrix of reservoir index values
+            res_pres(nr,ncell) = res_presx  ! matrix for if reservoir is present for each cell
+            rmile_node(ncell) = rmile1
+
```
Comment (YifanCheng) [inline src/Begin.f90:201]: Style problem again.
Within an `if` loop, you use 4 blanks in the `if` loop between L101 and L115 but you use 2 or 3 blanks in the `if` loop above. Meanwhile, there is one more blank in front of line 197 compared to line 194. I think they should match. You'd better read through the code and fix problems like these.

File: src/Block_Network.f90 [hunk 10]
```diff
@@ -21,4 +21,11 @@ Module Block_Network
     real   :: delta_n,n_default=2
     real   :: dt_comp
     real, dimension(:), allocatable  :: ndelta
+!
+!      Logical variables
+!
+    logical :: source  ! presence of thermal plant
+
+
+
```
Comment (YifanCheng) [inline src/Block_Network.f90:30]: Delete the three blank lines here or just leave one. There's no need to put so many blank lines here

File: src/Block_Reservoir.f90 [hunk 11]
```diff
@@ -1,28 +1,32 @@
...
+
         ! --------------------------------------------------------------------
         !       reservoir to RBM variables
         !----------------------------------------------------------------------
 
-        !------------------------read in reservoir info--------------------------
-        real, dimension (:), allocatable:: dam_lat, dam_lon, res_grid_lat, res_grid_lon
-        real, dimension (:), allocatable:: res_depth_feet, res_width_feet, res_length_feet
+        !------------------------read in reservoir
+        !info--------------------------
```
Comment (YifanCheng) [inline src/Block_Reservoir.f90]: Combine line 11 and line 12

File: src/Block_Reservoir.f90 [hunk 12]
```diff
@@ -1,28 +1,32 @@
...
+        real, dimension (:), allocatable:: dam_lat, dam_lon, res_grid_lat,res_grid_lon
+        real, dimension (:), allocatable:: res_depth_feet, res_width_feet,res_length_feet
         integer, dimension (:), allocatable:: dam_number, start_operating_year
-        real, dimension (:), allocatable:: res_top_vol, res_bot_vol, res_max_flow, res_min_flow
+        real, dimension (:), allocatable:: res_top_vol, res_bot_vol,res_max_flow, res_min_flow
         integer, dimension (:), allocatable:: res_start_node, res_end_node
-       !  integer, dimension (:,:), allocatable:: nodes_x   !for each reach, what are all the nodes
+       !  integer, dimension (:,:), allocatable:: nodes_x   !for each reach,
+       !  what are all the nodes
```
Comment (YifanCheng) [inline src/Block_Reservoir.f90]: In line 13, 14 and 16, add an blank between comma and next variable name.

Delete comment line 18 and line 19

File: src/Block_Reservoir.f90 [hunk 13]
```diff
@@ -1,28 +1,32 @@
...
-        logical, dimension(:), allocatable :: flag_turnover 
+        logical, dimension(:), allocatable :: flag_turnover
         logical :: reservoir, res_upstreamx ! the first is TRUE or FALSE in fifth line of _Network file whether reserovirs are present 
         integer, dimension (:), allocatable :: xres, resx
         integer :: xres2, nres, nm_start, ncell0res
-        real, dimension(:), allocatable :: dx_res, dt_res, reservoir_storage,reservoir_storage_prev
+        real, dimension(:), allocatable :: dx_res, dt_res,reservoir_storage,reservoir_storage_prev
         real ::  advec_tot,q_surf_tot
-        real, dimension (:), allocatable::diffusion_tot,advec_hyp_tot,advec_epi_tot, qsurf_tot
+        real, dimension (:),allocatable::diffusion_tot,advec_hyp_tot,advec_epi_tot, qsurf_tot
```
Comment (YifanCheng) [inline src/Block_Reservoir.f90]: In line 27, 29, 45, 47, 50, 52, 55, 56 and 65, add a blank between variable and comma.

File: src/Systmm.f90 [hunk 14]
```diff
@@ -60,6 +61,60 @@ SUBROUTINE SYSTMM(temp_file,param_file)
...
+allocate(temp_out_i(nres))
+temp_out_i = 10
+allocate (trib_res(heat_cells))
+allocate(diffusion_tot(nres))
+allocate(advec_hyp_tot(nres))
+allocate(advec_epi_tot(nres))
+allocate(qsurf_tot(nres))
+allocate(reservoir_storage(nres))
+allocate(reservoir_storage_prev(nres))
+
```
Comment (YifanCheng) [inline src/Systmm.f90:118]: It's better to put all those lines defining the initial values together. For example, you can put all those lines after you define all variables.

File: src/Systmm.f90 [hunk 15]
```diff
@@ -294,6 +395,10 @@ SUBROUTINE SYSTMM(temp_file,param_file)
 !   value of ndelta (now a vector)(UW_JRY_11/08/2013)
 !
             call WRITE(time,nd,nr,ncell,ns,T_0,T_head(nr),dbt(ncell),Q_inflow,Q_outflow)
+
+
```
Comment (YifanCheng) [inline src/Systmm.f90]: Delete one more blank lines

File: src/Systmm.f90 [hunk 16]
```diff
@@ -294,6 +395,10 @@ SUBROUTINE SYSTMM(temp_file,param_file)
...
+if(ncell .eq. 82)  write(67,*) nyear,',', ndays,',',nd,',',ncell,',',ns,',',T_0
```
Comment (YifanCheng) [inline src/Systmm.f90]: This is just an testing code. Do you want to leave it there?

**Comment by YifanCheng** [comment]
This pull request is outdated.

**Pull request closed without merging**